internal/service/imagebuilder/container_recipe_test.go

```diff
@@ -540,6 +541,40 @@ func TestAccImageBuilderContainerRecipe_kmsKeyID(t *testing.T) {
 	})
 }
 
+func TestAccImageBuilderContainerRecipe_platform_override(t *testing.T) {
+	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
+	resourceName := "aws_imagebuilder_container_recipe.test"
+	ecrImage := acctest.ImageBuilderECRImageFromEnv(t)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acctest.PreCheck(t) },
+		ErrorCheck:        acctest.ErrorCheck(t, imagebuilder.EndpointsID),
+		ProviderFactories: acctest.ProviderFactories,
+		CheckDestroy:      testAccCheckContainerRecipeDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccContainerRecipePlatformOverrideLinuxConfig(rName, ecrImage),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckContainerRecipeExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "platform_override", imagebuilder.PlatformLinux),
+				),
+			},
+			{
+				Config: testAccContainerRecipePlatformOverrideWindowsConfig(rName, ecrImage),
```

Comment:
I think you may need 2 variables; one to specify a custom Linux image, one to specify a custom Windows image. Perhaps: `IMAGE_BUILDER_ECR_IMAGE_LINUX` and `IMAGE_BUILDER_ECR_IMAGE_WINDOWS`, with `acctest.ImageBuilderECRImageFromEnv()` taking a second `suffix` arg (or similar) so it can check for, and spit out correct diagnostics for, both Linux & Windows images?

I'll give these tests a try out tomorrow once I've got the relevant ECR pre-reqs set up.

You may want to also consider a couple of negative test cases once this new test is working:

1. What happens if you specify `platform_override` when you're pointing at an Image Builder Image? I think you'll get an API error back, so deciding whether or not to implement some some plan-time validation to improve the quality of the feedback for users is up to us.
2. What happens if you specify a custom image but don't specify `platform_override`? Again, I suspect the provider will accept the request, only for it to be rejected by the API.

Reply:
My approach ended up being to spit the tests. However, I agree that probably having two variables makes sense.